src/prefect/orion/models/flow_runs.py

```diff
@@ -351,6 +352,7 @@ async def read_task_run_dependencies(
                 "state": task_run.state,
                 "expected_start_time": task_run.expected_start_time,
                 "name": task_run.name,
+                "display_name": task_run.task_key.rpartition(".")[-1],
```

Comment:
Why do we partition on a period?

Reply:
Also, if anything, I think the display name should be a portion of the task run name not the task key?

Reply:
The `task_key` is delimited by a `.`, in instances where there is no period, like when a user has defined a key manually, `rpartition` will still result in an array where the last value is the whole word.

Reply:
I see. The `task_key` defaults to the qualified import name of the task, a right-partition on a period does make sense here. However, the task key doesn't really make sense as the display name for a task run since task runs won't be distinguishable? A right partition of the `task_key` is the _same_ as the default value for the task name.

Reply:
That's correct. If there are more than one instance of a task run in a flow, it will show up twice with the same name, but the intent is for the UI to surface the unique name on click. The unique string at the end is not very human readable on it's own and clutters charts, making it more harmful than helpful when simply fallowing the series of events. 

You can see that demonstrated here:
https://user-images.githubusercontent.com/6776415/215178374-c839420b-5bbd-461b-9501-2a602846e9e9.mov

The individual task run name will be available when exploring the task details.